### DISCUSSION TITLE
[Q] Problem with "Implicit intersection operator: @"

### DISCUSSION THREAD

**Discussion by awn70** [exceljs/exceljs#2410] [Q&A]
I'm creating a simple sheet which includes an Excel array formula: ```SUM( ROUND( D1:D3, 0 ))```
With this code (ExcelJS 4.1.1):
```
     const workbook = new Excel.Workbook();
     const worksheet = workbook.addWorksheet( 'test_sheet' );
     worksheet.getCell( 'D1:D1' ).value = 1.2;
     worksheet.getCell( 'D2:D2' ).value = 2.3;
     worksheet.getCell( 'D3:D3' ).value = 5.6;
     worksheet.getCell( 'D4:D4' ).value = {
         formula: "SUM(ROUND(D1:D3, 0))",
         result: 0,
         date1904: false,
     };
     await workbook.xlsx.writeFile( "c:\\temp\\test.xlsx" );
```

When I open this sheet with Excel (Excel for Microsoft 365 MSO (16.0.13127.20266)) the new "[Implicit intersection operator](https://support.microsoft.com/en-us/office/implicit-intersection-operator-ce3be07b-0101-4450-a24e-c1c999be2b34)" will be inserted into the formula and breaks it:
![Sheet-with-iio](https://user-images.githubusercontent.com/1712315/95175737-35099a80-07bc-11eb-83e3-f474cf85b629.png)

Now I have to remove the inserted "@" and the formula works:
![Sheet-without-iio](https://user-images.githubusercontent.com/1712315/95175971-7e59ea00-07bc-11eb-893f-d206fbf69d23.png)

Has anybody a solution or workaround for this problem?

**Comment by alkafaiz** [comment]
same here. I use Excel for Microsoft 365
![image](https://user-images.githubusercontent.com/23336030/98078971-0dd2e580-1eae-11eb-94d4-6454fc490f5f.png)

**Comment by azael-leal** [comment]
I have the same problem using ExcelJS v4.1.1. I use Microsoft Excel for Mac (Microsoft 365, version 16.43 (20110804), language Español)

<img width="231" alt="Captura de Pantalla 2020-11-11 a la(s) 18 15 04" src="https://user-images.githubusercontent.com/55031967/98879128-63793600-244a-11eb-9f52-a9679ccbb43e.png">

**Comment by Russell-Tran** [comment]
Reporting the same issue for Microsoft Excel for Mac (Microsoft 365, version 16.46, language English). It breaks `CONCAT("a", "b")` by turning it into `@CONCAT("a", "b")`. A workaround is to use `CONCATENATE("a", "b")` instead.

**Comment by mcurran00** [comment]
I had the same issue with XLOOKUP. Entered a "good" version of formula in Excel and saved it. Read the file back through exceljs and the formula printed  as "_xlfn.XLOOKUP". When I specify the formula this way for exceljs it ends up avoiding putting the @ in Excel and the _xlfn is dropped by the time I look at it in Excel (and the formula works!). Same trick seems to work with CONCAT

**Comment by MikeIbberson** [comment]
The above solution worked in some cases, but for UNIQUE I had to also set a few other value properties. Putting "shareType" to "array" and providing "ref" did the trick. Mostly just commenting in case anyone else has similar issues with that function.

**Comment by verlok** [comment]
I filed a bug to this library.

In the meantime, @matteralus solution solved for me too.

**Comment by huberdo** [comment]
> The above solution worked in some cases, but for UNIQUE I had to also set a few other value properties. Putting "shareType" to "array" and providing "ref" did the trick. Mostly just commenting in case anyone else has similar issues with that function.

When I try to set the "shareType" to "array" I get: 
_Type '{ formula: string; result: number; shareType: string; ref: string; }' is not assignable to type 'CellValue'.
  Object literal may only specify known properties, and 'shareType' does not exist in type 'Date | CellErrorValue | CellRichTextValue | CellHyperlinkValue | CellFormulaValue | CellSharedFormulaValue'.ts(2322)_

Any ideas why (in TypeScript) shareType is not accessible?

Thank you for your help!

**Reply by bgdmrq** [reply]
same for me. did you find a solution for this?
